File: golang/06_list_of_strings/list_of_strings.go
package main

import "fmt"

func main() {
    var userstr string = ""
    userlist := []string{}
	
    for userstr != "-1" {
        fmt.Print("Enter a string (Enter -1 to quit): ")
	fmt.Scanf("%s", &userstr)
	userlist = append(userlist, userstr)
	fmt.Print(userlist, "\n")    
    }
}

File: rust/06_list_of_strings/list_of_strings.rs
fn main()
{
    let mut vec = Vec::new();
    
    loop {
        let mut inputstr = String::new();
        println!("Enter a new string (-1 to quit): ");
        let _ = std::io::stdin().read_line(&mut inputstr);
        inputstr.pop();  // remove trailing endline

        if inputstr == "-1"
	{
            break;
	}

        vec.push(inputstr.clone());
	println!("{:?}", vec)
    }
}